docs/ldml/tr35-keyboards.md

```diff
@@ -308,6 +308,279 @@ For extensibility, the `<special>` element will be allowed at nearly every level
 
 See [Element special](tr35.md#special) in Part 1.
 
+## Normalization
+
+Unicode Normalization, as described in [The Unicode Standard](https://www.unicode.org/reports/tr41/#Unicode/), is a process by which Unicode text is processed to eliminate unwanted distinctions.
+
+This section discusses how conformant keyboards are affected by normalization, and the impact of normalization on keyboard authors and keyboard implmentations.
+
+Keyboard implementation will usually apply normalization as appropriate when matching transform rules and `<display>` value matching.
+Output from the keyboard, following application of all transform rules, will be normalized to the appropriate form by the keyboard implementation.
+
+> Note: There are many existing software libraries which perform Unicode Normalization, including [ICU](https://icu.unicode.org), [ICU4X](https://icu4x.unicode.org), and JavaScript's [String.prototype.normalize()](https://developer.mozilla.org/docs/Web/JavaScript/Reference/Global_Objects/String/normalize).
+
+Keyboard authors will not typically need to perform normalization as part of the keyboard layout.  However, authors should be aware of areas where normalization affects keyboard operation so that they may achieve their desired results.
+
+### Where Normalization Occurs
+
+There are four stages where normalization occurs.
+
+1. **From the keyboard source `.xml`**
+
+    Keyboard source .xml files may be in any normalization form.
+
+    Example: `<key output=`, and `<transform from= to=` attribute contents
+    - From any form to NFD: full normalization (decompose+reorder)
+    - Markers must be processed
+    - Regex rules must be handled carefully
+
+2. **From the input context**
+
+    Input context must be normalized for purposes of matching.
+
+    Example: Input context might contain U+00E8 (`è`).  User clicks the cursor after the character. User types `<key ... output="\u{0320}"/>`.
```

Comment:
Same here: 
```suggestion
    Example: The input context contains U+00E8 (`è`).  The user clicks the cursor after the character, then types `<key ... output="\u{0320}"/>`.
```

Reply:
Also, do you really mean that the user literally types the characters `<, k, e, y, , ., ..., /, >`?

Reply:
The user types a key producing U+0320, thanks, will reword.